### PR TITLE
Fixes for coverty scan findings

### PR DESCRIPTION
**Pull request by ifranzki**
Closes: https://github.com/opencryptoki/opencryptoki/issues/562

### CODE REVIEW

**Review by juergenchrist — APPROVED** [review]

File: usr/lib/api/policy.c [hunk 1]
```diff
@@ -315,7 +315,7 @@ static CK_RV policy_get_sig_size(CK_MECHANISM_PTR mech, struct objstrength *s,
     const struct mechrow *col = mechrow_from_numeric(mech->mechanism);
     CK_ULONG size;
 
-    if (!col)
+    if (!col || !s)
```
[the remark this answers is not in the document]
Comment (ifranzki) [reply usr/lib/api/policy.c:318]: I would say no, but its hard to tell.
policy_get_sig_size() is for example called from policy_is_mech_allowed() dependent on the check type, passing the caller provded s.
policy_is_mech_allowed() is called with s = NULL, but as far as I can tell only for such checks that do not require s (i.e. for digests or keygen).

File: usr/lib/common/loadsave.c [hunk 2]
```diff
@@ -811,7 +811,12 @@ CK_RV load_private_token_objects_old(STDLL_TokData_t *tokdata)
...
             continue;
         }
-        //size--;
+        if (size <= sizeof(CK_ULONG_32) + sizeof(CK_BBOOL)) {
+            fclose(fp2);
+            OCK_SYSLOG(LOG_ERR, "Improper size of object %s (ignoring it)\n",
+                       fname);
+            continue;
+        }
         size = size - sizeof(CK_ULONG_32) - sizeof(CK_BBOOL);
```
Comment (holger-dengler) [inline usr/lib/common/loadsave.c:820]: It is not part of your change, but maybe you can also cover the next line as well

size -= sizeof(CK_ULONG_32) - sizeof(CK_BBOOL);

**Review by holger-dengler — APPROVED** [review]
Looks good to me. I approve this PR with or without my suggestions.

File: usr/lib/common/loadsave.c (hunk 2 above)
Comment (juergenchrist) [reply usr/lib/common/loadsave.c:820]: I think I know why this was not done.  Your version is wrong!!!  The second minus then has to be a plus.

Comment (holger-dengler) [reply usr/lib/common/loadsave.c:820]: You're right. It should be 
`size -= sizeof(CK_ULONG_32) + sizeof(CK_BBOOL);`

**Pull request merged**